internal/query/result_set.go

```diff
@@ -137,7 +137,7 @@ func (rs *resultSet) nextRow(ctx context.Context) (*row, error) {
 				}
 				rs.rowIndex = 0
 				rs.currentPart = part
-				if part == nil {
+				if part == nil || len(part.GetResultSet().GetRows()) == 0 {
```

Comment:
It can cause bug in the future.
Server can send statistics in middle of the stream (for example percent of progress), it can contains zero rows. 

What about continue to next iteration of for instead of return io.EOF?